src/acvp.c

```diff
@@ -1097,8 +1097,9 @@ ACVP_RESULT acvp_run_vectors_from_file(ACVP_CTX *ctx, const char *req_filename,
     /* If no vsIds found, try to get it from the filename */
     if (vs_cnt == 0) {
         int vsid;
+        int len = strnlen_s(req_filename, ACVP_JSON_FILENAME_MAX);
         char* start = strstr(req_filename, "vs");
-        if (!start || (0 == (vsid = atoi(start+2)))) {
+        if (!start || (start >= len+2) || (0 == (vsid = atoi(start+2)))) {
```

Comment:
Something doesn't feel right here. `start` is a memory address and `len` is a small length int. maybe `strnlen_s` should be run on `start`, and then check if `len <= 2` ?

Reply:
Good find, thanks.  Updated and ready.

Reply:
The acvpro_ prefix for strnlen_s does not exist in our code or compile - at the risk of being a pest :)

Reply:
Sorry!  Done